sql/sql_class.cc

```diff
@@ -3258,31 +3260,43 @@ static File create_file(THD *thd, char *path, sql_exchange *exchange,
     return -1;
   }
 
-  if (!access(path, F_OK))
+  if (my_stat(path, &file_stat, MYF(0)) && MY_S_ISFIFO(file_stat.st_mode))
   {
-    my_error(ER_FILE_EXISTS_ERROR, MYF(0), exchange->file_name);
-    return -1;
+    /* The specified file is a fifo */
+    if ((file= mysql_file_open(key_select_to_file,
+                               path, O_WRONLY, MYF(MY_WME))) < 0)
+      return file;
   }
-  /* Create the file world readable */
-  if ((file= mysql_file_create(key_select_to_file,
-                               path, 0644, O_WRONLY|O_EXCL, MYF(MY_WME))) < 0)
-    return file;
+  else
+  {
+    if (!access(path, F_OK))
+    {
+      my_error(ER_FILE_EXISTS_ERROR, MYF(0), exchange->file_name);
+      return -1;
+    }
+    /* Create the file world readable */
+    file_was_created_here= true;
+    if ((file= mysql_file_create(key_select_to_file,
+                                 path, 0644, O_WRONLY|O_EXCL, MYF(MY_WME))) < 0)
+      return file;
 #ifdef HAVE_FCHMOD
-  (void) fchmod(file, 0644);			// Because of umask()
+    (void) fchmod(file, 0644);			// Because of umask()
 #else
-  (void) chmod(path, 0644);
+    (void) chmod(path, 0644);
 #endif
+  }
+  
   if (init_io_cache(cache, file, 0L, WRITE_CACHE, 0L, 1, MYF(MY_WME)))
   {
     mysql_file_close(file, MYF(0));
-    /* Delete file on error, it was just created */
-    mysql_file_delete(key_select_to_file, path, MYF(0));
+    /* Delete file on error if it was just created */
+    if (file_was_created_here)
+      mysql_file_delete(key_select_to_file, path, MYF(0));
     return -1;
   }
   return file;
 }
 
```

Comment:
don't change unnecessary lines. The convention is 2 lines between functions.

Reply:
Sorry, that was accidental.  Restored.